identify-ref-leaks/README.md

```diff
@@ -0,0 +1,387 @@
+Here args for -f and -p are complusory where as -t is optional, by default
+time-interval is 15 minutes.
+Note : Time interval should be in minutes.
+
+And then script starts executing.
+
+After running the script, perform io’s. It should be taken care that the
+time interval should be large enough that all io’s are done in that.
+
+**4.1 What’s going on in background when stap script is running?**
+
+System tap executes the script in five rounds/passes.
+
+The stap command which is running in background looks something like
+this :
+
+stap -d /usr/local/lib/glusterfs/3.12dev/xlator/cluster/dht.so
+
+-d /usr/local/lib/glusterfs/3.12dev/xlator/cluster/afr.so
+
+-d /usr/local/lib/glusterfs/3.12dev/xlator/protocol/client.so
+
+-d /usr/local/lib/glusterfs/3.12dev/xlator/debug/io-stats.so
+
+-d /usr/local/lib/libgfrpc.so.0.0.1 -d
+/usr/local/lib/glusterfs/3.12dev/xlator/meta.so
+
+-d /usr/local/lib/glusterfs/3.12dev/rpc-transport/socket.so
+
+-d /usr/local/lib/glusterfs/3.12dev/xlator/mount/fuse.so
+
+-d /usr/lib64/libpthread-2.24.so -d /usr/lib64/libc-2.24.so
+
+-d /usr/local/lib/glusterfs/3.12dev/xlator/performance/io-threads.so
+
+-g --suppress-time-limits -DMAXSTRINGLEN=900 -S 5,100
+
+-o /var/run/gluster/leak-output/dict-leak1 -v /root/ex/test-dumb.stp
+
+-x 3860 dict\_ref dict\_unref 300
+
+Command uses certain options and arguments:
+
+-d : System tap requires information of these modules to be loaded into
+kernel space module. --suppress-time-limits : To disable MAXACTION ,
+which is maximum no. of statements to execute when a probe handler hits
+(default value : 1000) (It require guru mode i.e. -g)
+
+The rest options can be checked on stap man page. (man stap)
+
+**Concept of script** : In glusterfs when any process is using memory
+from memory pool, it simply calls ref/unref function, which keeps a
+count of no. of times it is being referenced or dereferenced. When this
+count is zero,this memory is put back into memory pool. The script is
+capturing this variable and storing tracebacks every time when it is
+used by any function. If the value is non-zero, it means that there
+would be a possible leak, which needs to be analyzed.
+
+Output : You will see the following when script runs :
+
+Pass 1: parsed user script and 479 library scripts using
+247904virt/50208res/7628shr/42752data kb, in 140usr/20sys/157real ms.
+
+Pass 2: analyzed script: 3 probes, 6 functions, 0 embeds, 7 globals
+using 251512virt/55104res/8688shr/46360data kb, in 30usr/0sys/24real ms.
+
+Pass 3: using cached
+/root/.systemtap/cache/a3/stap\_a3afa5f8e7d6cc1b8b20430003e866bb\_6194.c
+
+Pass 4: using cached
+/root/.systemtap/cache/a3/stap\_a3afa5f8e7d6cc1b8b20430003e866bb\_6194.ko
+
+Pass 5: starting run.
+
+**4.2 What are these Five passes?**
+
+The five passe flow can be understand easily by the following :
+
+![alt text](https://github.com/SonaArora/gluster-debug-tools/blob/master/systemtap-processing-steps.png){width="3.5989588801399823in"
```

Comment:
Is it really needed to have the full URL mentioned? Can you not just do it like

    [alt text](systemtap-processing-steps.png){...}

And I think `alt text` is expected to get replaced by a single sentence describing the image